functions/weather.location.fish

```diff
@@ -6,36 +6,34 @@ function weather.location -d "Get the current geographic location"
   end
 
   # Fetch location data based on our IP
-  if not set geoip_data (weather.fetch "http://geoip.nekudo.com/api/$ip")
+  if not set geoip_data (curl -s "https://ipapi.com/ip_api.php?ip=$ip")
     echo "Unable to query GeoIP data; please try again later."
     return 1
   end
-
+  
   # Echo coordiantes.
-  echo $geoip_data | jq '.location.latitude'
-  echo $geoip_data | jq '.location.longitude'
+  echo $geoip_data | jq '.latitude'
+  echo $geoip_data | jq '.longitude'
   echo $geoip_data | jq -r '.city?'
-  echo $geoip_data | jq -r '.country.name'
+  echo $geoip_data | jq -r '.country_name'
 end
 
 function __weather_get_ip -d "Get the current device's public IP address"
   # Attempt to get our external IP using the OpenDNS resolver.
-  if set -q __weather_system_dns
-    if not set ip (dig +short myip.opendns.com)
-      return 1
-    end
-  else
-    if not set ip (dig +short myip.opendns.com @resolver1.opendns.com)
-      return 1
-    end
-  end
+  # if set -q __weather_system_dns
+  #  if not set ip (dig +short myip.opendns.com)
+  #    return 1
+  #  end
+  # else
+  #  if not set ip (dig +short myip.opendns.com @resolver1.opendns.com)
+  #    return 1
+  #  end
+  # end
```

Comment:
Please undo all this.

Reply:
This is broken, it needs to be removed. Commented as a courtesy.

Reply:
I agree with Scorphus. This is a bad smell for code. We should either figure it out what's going on or just delete it, after all, anyone can find this code in the history.

I'm not familiar with shell development. If you can tell me how to run this code locally, maybe I can help.

Reply:
Not only it isn't broken, getting our external IP through `dig` and OpenDNS is way faster:

```
»»»» time dig +short myip.opendns.com @resolver1.opendns.com
109.41.193.125
        0.05 real         0.00 user         0.00 sys
```

ipecho.net is there as a fallback:

```
»»»» time curl -s http://ipecho.net/plain
123.12.23.123
        0.27 real         0.00 user         0.00 sys
```